libparsec/crates/serialization_format/src/protocol_tests.rs

```diff
@@ -0,0 +1,95 @@
+// Parsec Cloud (https://parsec.cloud) Copyright (c) BUSL-1.1 2016-present Scille SAS
+
+use proc_macro2::TokenStream;
+use quote::{format_ident, quote};
+
+use super::*;
+
+/// See [crate::generate_protocol_cmds_family_from_contents]
+pub(crate) fn generate_protocol_cmds_tests(cmds: Vec<JsonCmd>, family_name: &str) -> TokenStream {
+    let family = &GenCmdsFamily::new(cmds, family_name, false);
+    let family_name = format_ident!("{}", &family.name);
+    let versioned_cmds = family
+        .versions
+        .iter()
+        .sorted_by_key(|(v, _)| *v)
+        .map(|(version, cmds)| quote_versioned_cmds_test(*version, family, cmds));
+
+    quote! {
+        pub mod #family_name {
+            #(#versioned_cmds)*
+        }
+    }
+}
+
+fn quote_versioned_cmds_test(version: u32, family: &GenCmdsFamily, cmds: &[GenCmd]) -> TokenStream {
+    let mod_version = format_ident!("v{version}");
+    let family_name = format_ident!("{}", &family.name);
+    let cmd_tests = cmds.iter().map(|cmd| quote_cmd_tests(version, cmd));
+
+    quote! {
+        pub mod #mod_version {
+            use libparsec_protocol::#family_name::#mod_version as #family_name;
```

Comment:
We should add a comment here to stress there is a trick here, something like:
```rust
// The secret sauce is here ! cmd module accesses the schema by doing
// e.g. `use super::authenticated_cmds` (instead of the regular `libparsce_protocol::authenticated_cmd::vX`)
// This allows to reuse the cmd module multiple times (i.e. not `use` it but actually compiling
// it), each time with a parent module defining a different version of the familly command

Reply:
Done